taichi/runtime/llvm/runtime.cpp

```diff
@@ -88,7 +88,24 @@ void locked_task(void *lock, const T &func);
 template <typename T>
 T ifloordiv(T a, T b) {
   auto r = a / b;
-  r -= T(a * b < 0 && b * r != a);
+  // simply `a * b < 0` may leads to overflow (#969)
+  //
+  // Formal Anti-Regression Verification (FARV):
+  //
+  // old = a * b < 0
+  // new = (a < 0) != (b < 0) && a
+  //
+  //  a  b old new
+  //  -  -  f = f (f&f)
```

Comment:
Nice comment! I think you forgot to update the rhs's value?

```suggestion
  //  -  -  f = f (f&t)
  //  -  +  t = t (t&t)
  //  0  -  f = f (t&f)
  //  0  +  f = f (f&f)
  //  +  -  t = t (t&t)
  //  +  +  f = f (f&t)
```